featurebyte/api/use_case.py

```diff
@@ -55,30 +56,40 @@ class UseCase(SavableApiObject, DeletableApiObject):
     ]
 
     # pydantic instance variable (public)
-    target: Target
-    # Context api object is not implemented yet. This is a placeholder.
-    context: Optional[Any]
+    target_id: PydanticObjectId
+    context_id: PydanticObjectId
 
-    def _get_create_payload(self) -> dict[str, Any]:
+    @property
+    def target(self) -> Target:
         """
-        Get the payload for creating a new Use Case.
+        Returns the target object of the UseCase.
 
         Returns
         -------
-        dict[str, Any]
+        Target
+            The target object of the UseCase.
         """
-        data = UseCaseCreate(**self.dict(by_alias=True), target_id=self.target.id)
-        return data.json_dict()
+        return Target.get_by_id(self.target_id)
 
-    def _get_init_params_from_object(self) -> dict[str, Any]:
-        return {"target": self.target}
+    @property
+    def context(self) -> Context:
+        """
+        Returns the context object of the UseCase.
+
+        Returns
+        -------
+        Context
+            The context object of the UseCase.
+        """
+        return Context.get_by_id(self.context_id)
 
     @classmethod
+    @typechecked
     def create(
         cls,
         name: str,
         target: Target,
-        context: Optional[Any] = None,
+        context: Context,
```

Comment:
similar to other PR, i feel like we normally take in names of the objects we're creating here in the rest of the SDK.

Reply:
ok. will change to take in names instead